rust_dev_preview/dynamodb/src/scenario/create.rs

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_sdk_dynamodb::model::{
+    AttributeDefinition, AttributeValue, KeySchemaElement, KeyType, ProvisionedThroughput,
+    ScalarAttributeType,
+};
+use aws_sdk_dynamodb::{Client, Error};
+
+// Create a table.
+// snippet-start:[dynamodb.rust.create-table]
+pub async fn create_table(client: &Client, table: &str, key: &str) -> Result<(), Error> {
+    let a_name: String = key.into();
+    let table_name: String = table.into();
+
+    let ad = AttributeDefinition::builder()
+        .attribute_name(&a_name)
+        .attribute_type(ScalarAttributeType::S)
+        .build();
+
+    let ks = KeySchemaElement::builder()
+        .attribute_name(&a_name)
+        .key_type(KeyType::Hash)
+        .build();
+
+    let pt = ProvisionedThroughput::builder()
+        .read_capacity_units(10)
+        .write_capacity_units(5)
+        .build();
+
+    let create_table_response = client
+        .create_table()
+        .table_name(table_name)
+        .key_schema(ks)
+        .attribute_definitions(ad)
+        .provisioned_throughput(pt)
+        .send()
+        .await;
+
+    match create_table_response {
+        Ok(_) => {
+            println!("Added table {} with key {}", table, key);
+            Ok(())
+        }
+        Err(e) => {
+            eprintln!("Got an error creating table:");
+            eprintln!("{}", e);
+            Err(Error::Unhandled(Box::new(e)))
+        }
+    }
+}
+// snippet-end:[dynamodb.rust.create-table]
+
+pub struct Item {
+    pub p_type: String,
+    pub age: String,
+    pub username: String,
+    pub first: String,
+    pub last: String,
+}
+
+// Add an item to a table.
+// snippet-start:[dynamodb.rust.add-item]
+pub async fn add_item(client: &Client, item: Item, table: &String) -> Result<(), Error> {
```

Comment:
Seems like `add_item` should be in its own scenario file. I don't think I would check `create` for it unless I looked at the import for it while looking at one of the `main` functions.

Reply:
I was naming the scenario mods after CRUD, but I have List, so might as well move it to add.rs